sonar-batch/src/main/java/org/sonar/batch/report/MeasuresPublisher.java

```diff
@@ -86,13 +87,18 @@ private void setValueAccordingToType(BatchReport.Measure.Builder builder, Measur
         builder.setBooleanValue((Boolean) value);
         break;
       case DOUBLE:
-        builder.setDoubleValue((Double) value);
+        builder.setDoubleValue(((Number) value).doubleValue());
         break;
       case INT:
-        builder.setIntValue((Integer) value);
+        builder.setIntValue(((Number) value).intValue());
         break;
       case LONG:
-        builder.setLongValue((Long) value);
+        try {
+          builder.setLongValue(((Number) value).longValue());
+        } catch (RuntimeException e) {
+          System.out.println(Objects.toString(measure));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS106)